{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "streamlittest3.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "authorship_tag": "ABX9TyPVF1SNMQ9vKj1d2iLF0V0P",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/AryanWadhwa05/Stock-Analysis-and-Prediction/blob/main/streamlit.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tEe7t7FpUf3O",
        "outputId": "2440753f-26a4-4576-965a-3ed52dcddff8"
      },
      "source": [
        "%%writefile app.py\n",
        "import streamlit as st\n",
        "import yfinance as yf\n",
        "import pandas as pd\n",
        "import matplotlib.pyplot as plt\n",
        "from pandas_datareader import data as pdr\n",
        "\n",
        "\n",
        "# download dataframe\n",
        "st.title(\"Stock Market Analysis\")\n",
        "\n",
        "#sidebar\n",
        "\n",
        "# st.file_uploader(\"Upload CSV\")\n",
        "\n",
        "st.sidebar.title(\"Stocks\")\n",
        "\n",
        "st.sidebar.subheader(\"Stock Search\")\n",
        "\n",
        "# st.sidebar.selectbox(\"Example\",[\"Facebook\",\"Apple\",\"Alphabet\"])\n",
        "stock_search = st.sidebar.multiselect('Stock Search', [\"FB\",\"AAPL\",\"Alphabet\"],default = \"FB\")\n",
        "display_cols = st.sidebar.multiselect('Display Data', [\"Close\",\"Adj Close\",\"Low\",\"Open\",\"High\",\"Volume\"],default = \"High\")\n",
        "graph_type = st.sidebar.selectbox('Graph Type', [\"Line\",\"Bar\",\"Vega Lite\",\"AltAir\",\"High\",\"Volume\"])\n",
        "\n",
        "start_date = st.sidebar.date_input('Start Date')\n",
        "end_date = st.sidebar.date_input('End Date')\n",
        "\n",
        "# st.write(start_date)\n",
        "\n",
        "yf.pdr_override()\n",
        "tickers = \"\"\n",
        "for i in stock_search:\n",
        "  tickers += i + \" \"\n",
        "\n",
        "st.write(tickers)\n",
        "# sample_data=pdr.get_data_yahoo(stock_search[i], start=start_date, end=end_date)\n",
        "data = yf.download(tickers, start=start_date, end=end_date)\n",
        "st.write(data)\n",
        "\n",
        "dataframe = data[display_cols]\n",
        "st.write(dataframe)\n",
        "st.line_chart(dataframe)\n",
        "\n",
        "# plt.plot(dataframe)\n",
        "    \n",
        "# st.pyplot(plt)\n",
        "\n",
        "# data1 = pdr.get_data_yahoo(\"FB\", start=start_date, end=end_date)\n",
        "\n",
        "# dataframe = pd.DataFrame()\n",
        "\n",
        "# for i in stock_search:\n",
        "#   dataframe[i]\n",
        "#   data=yf.download(i, start=start_date, end=end_date)\n",
        "#   dataframe[i] = data[\"High\"]\n",
        "\n",
        "# st.write(dataframe)\n",
        "# dataframe1\n",
        "\n",
        "# dataframe1 = data1[\"Low\"]\n",
        "# dataframe1 = dataframe1[\"Apple\"]\n",
        "# dataframe1[\"Apple\"] = data[\"Low\"]\n",
        "\n",
        "# data_frame = data[display_cols]\n",
        "# data = data.drop([\"Volume\",\"Close\",\"Adj Close\",\"Low\",\"Open\"],axis = 1)\n",
        "\n",
        "# st.pyplot(data)\n",
        "\n",
        "# st.line_chart(dataframe)"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Writing app.py\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aKOXO1vVUm5M",
        "outputId": "638b585d-2dd5-495f-b36d-c2a50936a3e0"
      },
      "source": [
        "#@title Run this cell to load some data\n",
        "!pip -q install streamlit > /dev/null\n",
        "!pip -q install pyngrok > /dev/null\n",
        "!pip install yfinance\n",
        "\n",
        "import tensorflow as tf\n",
        "import pandas as pd\n",
        "import cv2\n",
        "import plotly.express as px\n",
        "import numpy as np\n",
        "import yfinance as yf"
      ],
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "jupyter-console 5.2.0 requires prompt-toolkit<2.0.0,>=1.0.0, but you have prompt-toolkit 3.0.20 which is incompatible.\n",
            "google-colab 1.0.0 requires ipykernel~=4.10, but you have ipykernel 6.3.1 which is incompatible.\n",
            "google-colab 1.0.0 requires ipython~=5.5.0, but you have ipython 7.27.0 which is incompatible.\u001b[0m\n",
            "Collecting yfinance\n",
            "  Downloading yfinance-0.1.63.tar.gz (26 kB)\n",
            "Requirement already satisfied: pandas>=0.24 in /usr/local/lib/python3.7/dist-packages (from yfinance) (1.1.5)\n",
            "Requirement already satisfied: numpy>=1.15 in /usr/local/lib/python3.7/dist-packages (from yfinance) (1.19.5)\n",
            "Requirement already satisfied: requests>=2.20 in /usr/local/lib/python3.7/dist-packages (from yfinance) (2.23.0)\n",
            "Requirement already satisfied: multitasking>=0.0.7 in /usr/local/lib/python3.7/dist-packages (from yfinance) (0.0.9)\n",
            "Collecting lxml>=4.5.1\n",
            "  Downloading lxml-4.6.3-cp37-cp37m-manylinux2014_x86_64.whl (6.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 6.3 MB 5.9 MB/s \n",
            "\u001b[?25hRequirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.7/dist-packages (from pandas>=0.24->yfinance) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.7/dist-packages (from pandas>=0.24->yfinance) (2018.9)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/dist-packages (from python-dateutil>=2.7.3->pandas>=0.24->yfinance) (1.15.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20->yfinance) (2021.5.30)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20->yfinance) (1.24.3)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20->yfinance) (3.0.4)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20->yfinance) (2.10)\n",
            "Building wheels for collected packages: yfinance\n",
            "  Building wheel for yfinance (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for yfinance: filename=yfinance-0.1.63-py2.py3-none-any.whl size=23918 sha256=7b8931798fa36e6cecfd49853362b9e8f2651b2da57c9040d7d565b5556fbe49\n",
            "  Stored in directory: /root/.cache/pip/wheels/fe/87/8b/7ec24486e001d3926537f5f7801f57a74d181be25b11157983\n",
            "Successfully built yfinance\n",
            "Installing collected packages: lxml, yfinance\n",
            "  Attempting uninstall: lxml\n",
            "    Found existing installation: lxml 4.2.6\n",
            "    Uninstalling lxml-4.2.6:\n",
            "      Successfully uninstalled lxml-4.2.6\n",
            "Successfully installed lxml-4.6.3 yfinance-0.1.63\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Cu7XXyy4UkCX",
        "outputId": "f767c9ed-e755-4415-c785-a469ea4b2762"
      },
      "source": [
        "from pyngrok import ngrok\n",
        "#Publish Web App (Run this again whenever you make changes)\n",
        "public_url = ngrok.connect(port='80')\n",
        "print (public_url)\n",
        "!streamlit run --server.port 80 app.py"
      ],
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "NgrokTunnel: \"http://b578-35-196-106-72.ngrok.io\" -> \"http://localhost:80\"\n",
            "\u001b[0m\n",
            "\u001b[34m\u001b[1m  You can now view your Streamlit app in your browser.\u001b[0m\n",
            "\u001b[0m\n",
            "\u001b[34m  Network URL: \u001b[0m\u001b[1mhttp://172.28.0.2:80\u001b[0m\n",
            "\u001b[34m  External URL: \u001b[0m\u001b[1mhttp://35.196.106.72:80\u001b[0m\n",
            "\u001b[0m\n",
            "2021-09-09 02:12:46.656 NumExpr defaulting to 2 threads.\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  3 of 3 completed\n",
            "\n",
            "1 Failed download:\n",
            "- P: No data found for this date range, symbol may be delisted\n",
            "2021-09-09 02:14:05.096 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 42, in <module>\n",
            "    st.line_chart(dataframe)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/dataframe_selector.py\", line 166, in line_chart\n",
            "    return self.dg._arrow_line_chart(data, width, height, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 92, in _arrow_line_chart\n",
            "    marshall(proto, chart, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 369, in marshall\n",
            "    chart_dict = altair_chart.to_dict()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/api.py\", line 373, in to_dict\n",
            "    dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in to_dict\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in <listcomp>\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 42, in to_dict\n",
            "    \"match any column in the data.\".format(shorthand))\n",
            "ValueError: variable encoding field is specified without a type; the type cannot be inferred because it does not match any column in the data.\n",
            "\n",
            "[*********************100%***********************]  5 of 5 completed\n",
            "\n",
            "1 Failed download:\n",
            "- P: No data found for this date range, symbol may be delisted\n",
            "2021-09-09 02:14:21.989 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 42, in <module>\n",
            "    st.line_chart(dataframe)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/dataframe_selector.py\", line 166, in line_chart\n",
            "    return self.dg._arrow_line_chart(data, width, height, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 92, in _arrow_line_chart\n",
            "    marshall(proto, chart, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 369, in marshall\n",
            "    chart_dict = altair_chart.to_dict()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/api.py\", line 373, in to_dict\n",
            "    dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in to_dict\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in <listcomp>\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 42, in to_dict\n",
            "    \"match any column in the data.\".format(shorthand))\n",
            "ValueError: variable encoding field is specified without a type; the type cannot be inferred because it does not match any column in the data.\n",
            "\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "2021-09-09 02:23:08.944 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 45, in <module>\n",
            "    st.line_chart(dataframe)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/dataframe_selector.py\", line 166, in line_chart\n",
            "    return self.dg._arrow_line_chart(data, width, height, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 92, in _arrow_line_chart\n",
            "    marshall(proto, chart, use_container_width)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/arrow_altair.py\", line 369, in marshall\n",
            "    chart_dict = altair_chart.to_dict()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/api.py\", line 373, in to_dict\n",
            "    dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 328, in to_dict\n",
            "    context=context,\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 62, in _todict\n",
            "    for k, v in obj.items()\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 63, in <dictcomp>\n",
            "    if v is not Undefined\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/utils/schemapi.py\", line 56, in _todict\n",
            "    return obj.to_dict(validate=validate, context=context)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in to_dict\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 25, in <listcomp>\n",
            "    for sh in shorthand]\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/altair/vegalite/v4/schema/channels.py\", line 42, in to_dict\n",
            "    \"match any column in the data.\".format(shorthand))\n",
            "ValueError: variable encoding field is specified without a type; the type cannot be inferred because it does not match any column in the data.\n",
            "\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "[*********************100%***********************]  3 of 3 completed\n",
            "[*********************100%***********************]  4 of 4 completed\n",
            "[*********************100%***********************]  5 of 5 completed\n",
            "[*********************100%***********************]  6 of 6 completed\n",
            "[*********************100%***********************]  7 of 7 completed\n",
            "\n",
            "1 Failed download:\n",
            "- MDAIJ: No data found, symbol may be delisted\n",
            "[*********************100%***********************]  7 of 7 completed\n",
            "\n",
            "1 Failed download:\n",
            "- MDAIJ: No data found, symbol may be delisted\n",
            "[*********************100%***********************]  7 of 7 completed\n",
            "\n",
            "1 Failed download:\n",
            "- MDAIJ: No data found, symbol may be delisted\n",
            "2021-09-09 02:43:24.459 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"year\", y=\"price\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 946, in build_dataframe\n",
            "    raise ValueError(err_msg)\n",
            "ValueError: Value of 'x' is not the name of a column in 'data_frame'. Expected one of [('High', 'AAPL'), ('High', 'FB'), ('High', 'M'), ('High', 'MC'), ('High', 'MDAIJ'), ('High', 'MS'), ('High', 'MSFT')] but received: year\n",
            "\n",
            "[*********************100%***********************]  6 of 6 completed\n",
            "2021-09-09 02:43:47.937 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"year\", y=\"price\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 946, in build_dataframe\n",
            "    raise ValueError(err_msg)\n",
            "ValueError: Value of 'x' is not the name of a column in 'data_frame'. Expected one of [('High', 'AAPL'), ('High', 'FB'), ('High', 'M'), ('High', 'MC'), ('High', 'MS'), ('High', 'MSFT')] but received: year\n",
            "\n",
            "[*********************100%***********************]  6 of 6 completed\n",
            "2021-09-09 02:44:03.729 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"High\", y=\"price\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 960, in build_dataframe\n",
            "    df_output[col_name] = df_input[argument].values\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3044, in __setitem__\n",
            "    self._set_item(key, value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3119, in _set_item\n",
            "    self._ensure_valid_index(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3170, in _ensure_valid_index\n",
            "    value = Series(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/series.py\", line 327, in __init__\n",
            "    data = sanitize_array(data, index, dtype, copy, raise_cast_failure=True)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/construction.py\", line 496, in sanitize_array\n",
            "    raise Exception(\"Data must be 1-dimensional\")\n",
            "Exception: Data must be 1-dimensional\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "2021-09-09 02:44:13.538 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"High\", y=\"price\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 946, in build_dataframe\n",
            "    raise ValueError(err_msg)\n",
            "ValueError: Value of 'y' is not the name of a column in 'data_frame'. Expected one of ['High'] but received: price\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "2021-09-09 02:44:27.674 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"year\", y=\"High\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 946, in build_dataframe\n",
            "    raise ValueError(err_msg)\n",
            "ValueError: Value of 'x' is not the name of a column in 'data_frame'. Expected one of ['High'] but received: year\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "2021-09-09 02:46:15.322 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe, x=\"year\", y=\"High\", title='Stock Market Data')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 946, in build_dataframe\n",
            "    raise ValueError(err_msg)\n",
            "ValueError: Value of 'x' is not the name of a column in 'data_frame'. Expected one of ['High'] but received: year\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "2021-09-09 02:48:49.669 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 47, in <module>\n",
            "    fig = px.line(dataframe,y=\"High\")\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_chart_types.py\", line 214, in line\n",
            "    return make_figure(args=locals(), constructor=go.Scatter)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1172, in make_figure\n",
            "    args, constructor, trace_patch\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 1028, in infer_config\n",
            "    args = build_dataframe(args, all_attrables, array_attrables)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/plotly/express/_core.py\", line 960, in build_dataframe\n",
            "    df_output[col_name] = df_input[argument].values\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3044, in __setitem__\n",
            "    self._set_item(key, value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3119, in _set_item\n",
            "    self._ensure_valid_index(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py\", line 3170, in _ensure_valid_index\n",
            "    value = Series(value)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/series.py\", line 327, in __init__\n",
            "    data = sanitize_array(data, index, dtype, copy, raise_cast_failure=True)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/pandas/core/construction.py\", line 496, in sanitize_array\n",
            "    raise Exception(\"Data must be 1-dimensional\")\n",
            "Exception: Data must be 1-dimensional\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "2021-09-09 02:49:53.888 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 53, in <module>\n",
            "    st.pyplot(fig)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/pyplot.py\", line 93, in pyplot\n",
            "    self.dg._get_delta_path_str(), image_list_proto, fig, clear_figure, **kwargs\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/pyplot.py\", line 132, in marshall\n",
            "    fig.savefig(image, **kwargs)\n",
            "AttributeError: 'Figure' object has no attribute 'savefig'\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "2021-09-09 02:51:46.227 Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/script_runner.py\", line 354, in _run_script\n",
            "    exec(code, module.__dict__)\n",
            "  File \"/content/app.py\", line 53, in <module>\n",
            "    st.pyplot(fig)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/pyplot.py\", line 93, in pyplot\n",
            "    self.dg._get_delta_path_str(), image_list_proto, fig, clear_figure, **kwargs\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/streamlit/elements/pyplot.py\", line 132, in marshall\n",
            "    fig.savefig(image, **kwargs)\n",
            "AttributeError: 'Figure' object has no attribute 'savefig'\n",
            "\n",
            "[*********************100%***********************]  1 of 1 completed\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "[*********************100%***********************]  2 of 2 completed\n",
            "\u001b[34m  Stopping...\u001b[0m\n",
            "\u001b[34m  Stopping...\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "950OzZl-gOzl"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6igcuC_vgO5z"
      },
      "source": [
        "%%writefile app.py\n",
        "\n",
        "import streamlit as st\n",
        "import yfinance as yf\n",
        "import pandas as pd\n",
        "import matplotlib.pyplot as plt\n",
        "from pandas_datareader import data as pdr\n",
        "import plotly.express as px\n",
        "\n",
        "# download dataframe\n",
        "st.title(\"Stock Market Analysis\")\n",
        "\n",
        "#sidebar\n",
        "\n",
        "# st.file_uploader(\"Upload CSV\")\n",
        "\n",
        "st.sidebar.title(\"Stocks\")\n",
        "\n",
        "st.sidebar.subheader(\"Stock Search\")\n",
        "\n",
        "# st.sidebar.selectbox(\"Example\",[\"Facebook\",\"Apple\",\"Alphabet\"])\n",
        "# stock_search = st.sidebar.multiselect('Stock Search', [\"FB\",\"AAPL\",\"Alphabet\"],default = \"FB\")\n",
        "stock_search_test = st.sidebar.text_input('Stock Search')\n",
        "display_cols = st.sidebar.multiselect('Display Data', [\"Close\",\"Adj Close\",\"Low\",\"Open\",\"High\",\"Volume\"],default = \"High\")\n",
        "graph_type = st.sidebar.selectbox('Graph Type', [\"Line\",\"Bar\",\"Vega Lite\",\"AltAir\",\"High\",\"Volume\"])\n",
        "\n",
        "start_date = st.sidebar.date_input('Start Date')\n",
        "end_date = st.sidebar.date_input('End Date')\n",
        "\n",
        "# st.write(start_date)\n",
        "\n",
        "yf.pdr_override()\n",
        "tickers = \"\"\n",
        "\n",
        "tickers = stock_search_test.replace(\",\",\" \")\n",
        "\n",
        "# for i in stock_search_test:\n",
        "  # tickers += i + \" \"\n",
        "\n",
        "# st.write(tickers)\n",
        "# sample_data=pdr.get_data_yahoo(stock_search[i], start=start_date, end=end_date)\n",
        "data = yf.download(tickers, start=start_date, end=end_date)\n",
        "# st.write(data)\n",
        "\n",
        "dataframe = data[display_cols]\n",
        "st.write(dataframe)\n",
        "# st.line_chart(dataframe)\n",
        "\n",
        "# fig = px.line(dataframe,y=\"High\")\n",
        "# fig.show()\n",
        "# st.plotly_chart(fig)\n",
        "\n",
        "plt.plot(dataframe)\n",
        "plt.legend(dataframe)\n",
        "\n",
        "st.pyplot(plt)\n",
        "\n",
        "# data1 = pdr.get_data_yahoo(\"FB\", start=start_date, end=end_date)\n",
        "\n",
        "# dataframe = pd.DataFrame()\n",
        "\n",
        "# for i in stock_search:\n",
        "#   dataframe[i]\n",
        "#   data=yf.download(i, start=start_date, end=end_date)\n",
        "#   dataframe[i] = data[\"High\"]\n",
        "\n",
        "# st.write(dataframe)\n",
        "# dataframe1\n",
        "\n",
        "# dataframe1 = data1[\"Low\"]\n",
        "# dataframe1 = dataframe1[\"Apple\"]\n",
        "# dataframe1[\"Apple\"] = data[\"Low\"]\n",
        "\n",
        "# data_frame = data[display_cols]\n",
        "# data = data.drop([\"Volume\",\"Close\",\"Adj Close\",\"Low\",\"Open\"],axis = 1)\n",
        "\n",
        "# st.pyplot(data)\n",
        "\n",
        "# st.line_chart(dataframe)"
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}